src/analyitcs/provider/analytics.provider.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from 'react'
+
+import fathomSettings from '../config/fathom-settings.json'
+import { loadFathom } from '../fathom'
+import { FathomRegisteredEvent, FathomSite } from '../types'
+import { AnalyticsContext, IAnalyticsContext } from './analytics.context'
+
+interface AnalyticsProviderProps {
+  children: React.ReactNode
+}
+
+/**
+ * AnalyticsProvider: provides the analytics context to the application
+ */
+export function AnalyticsProvider({ children }: AnalyticsProviderProps) {
+  const [site, setSite] = useState<FathomSite>()
+
+  useEffect(() => {
+    const siteId = process.env.REACT_APP_FATHOM_SITE_ID
+
+    if (!siteId) {
+      if (process.env.NODE_ENV === 'development') {
+        console.warn('REACT_APP_FATHOM_SITE_ID not set, skipping Fathom analytics')
+      }
+      return
+    }
+
+    const siteDetails = fathomSettings.fathomSites.find(site => site.siteId === siteId)
+
+    if (!siteDetails) {
+      console.error(`Fathom site ID (${siteId}) not found in fathom-settings.json`)
+      return
+    }
+
+    loadFathom(siteId)
+      .then(() => {
+        setSite(siteDetails)
+      })
+      .catch(error => {
+        console.error('Error loading Fathom analytics', error)
+      })
+  }, [])
+
+  const context = {
+    site,
+    fathom: window.fathom,
+    trackRegisteredEvent: (registeredEventName: FathomRegisteredEvent, data: Record<string, any>) => {
+      if (!site || !window.fathom) {
+        return
+      }
+      const eventDetails = site.events.find(
+        event => event.name.toLocaleUpperCase() === registeredEventName.toLocaleLowerCase()
+      )
+      if (!eventDetails) {
+        return
+      }
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      window.fathom.trackEvent(eventDetails.id, data)
```

Comment:
I'm wondering how does fhantom tracks the transaction volume?

Reply:
I updated the PR with more details. Fathom has a counter model. We send the new value for the named counter (see the PR for named keys), and Fathom increases it.

Reply:
Yup, I've checked the new code, I was looking for this code 👇

<img width="527" alt="image" src="https://user-images.githubusercontent.com/5664434/190664576-1cbf3abb-320f-4da7-a0b4-930a13443004.png">


https://github.com/levelkdev/swapr-dapp/pull/1476/files#diff-686eddb0e7f6457b55065661febd02eddda11d03b5350f62fd7dc13f275a08d0R201

Much clear now. Great job Adam 🙌